### ISSUE TITLE
Cluster resources not cleaned up after tracking via annotation+label

### ISSUE THREAD

**Issue by amottasr** [argoproj/argo-cd#15101]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When setting `application.resourceTrackingMethod: annotation+label`, cluster resources are not cleaned up after the application is deleted.  Additionally, we have noticed that a namespace is still added to the annotations of cluster resources:
```yaml
metadata:
  annotations:
    argocd.argoproj.io/tracking-id: kyverno:rbac.authorization.k8s.io/ClusterRole:kyverno/kyverno:webhook
```

**To Reproduce**
1. Set `application.resourceTrackingMethod: annotation+label`
2. Delete some argocd application that deploys cluster resources
3. Notice that cluster resources remain after application is deleted

**Expected behavior**

Cluster resources that are tracked with an application are also cleaned up once that application is deleted  

**Version**

```shell
argocd-server: v2.6.3+e05298b
```

**Comment by oreonl** [comment]
Has anyone had a chance to look at this issue?

**Comment by oreonl** [comment]
Digging a little more into this it appears to be specific to resources that have a colon in the name. This is not legal for most resources, but RBAC resources allow it. [This](https://github.com/argoproj/argo-cd/blob/eba40d4dec9b88a7d4cdee5d645193a6aaa0cf47/util/argo/resource_tracking.go#L181-L185) seems to be the relevant code. It is splitting the annotation by `:` and expecting a length of 3. So any resources with extra `:` in the name cause it to return nil. Could the delimiter be updated to be something that is not allowed in any resource names? Or be able to handle extra colons?

**Comment by blakepettersson** [comment]
Fixed with #15380
